File: app/src/main/java/mx/dev/franco/automusictagfixer/di/TaggerModule.java
package mx.dev.franco.automusictagfixer.di;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import mx.dev.franco.automusictagfixer.fixer.AudioTagger;

@Module
public class TaggerModule {

    @Singleton
    @Provides
    AudioTagger providesTagger(Application application, AudioTagger.StorageHelper storageHelper) {
        return new AudioTagger(application, storageHelper);
    }
}

File: app/src/main/java/mx/dev/franco/automusictagfixer/di/TaggerModule.kt
package mx.dev.franco.automusictagfixer.di

import android.app.Application
import dagger.Module
import dagger.Provides
import mx.dev.franco.automusictagfixer.fixer.AudioTagger
import mx.dev.franco.automusictagfixer.fixer.StorageHelper
import javax.inject.Singleton

@Module
class TaggerModule {

    @Singleton
    @Provides
    fun providesTagger(application: Application, storageHelper: StorageHelper) =
        AudioTagger(application, storageHelper)

}